Task#3/main.cpp

```diff
@@ -0,0 +1,65 @@
+#include <iostream>
+#include "filestat.h"
+#include <QDir>
+#include <thread>
+#include <QThreadPool>
+
+QFileInfoList recursiveSearch(QString project_path) {
+    QDir project(project_path);
+    QFileInfoList elements(project.entryInfoList());
+    QFileInfoList files;
```

Comment:
You might pass to this method additional argument - file filter to pick files with specific extensions

QFileInfoList QDir::entryInfoList(const QStringList &nameFilters, _**QDir::Filters filters = NoFilter**_, QDir::SortFlags sort = NoSort) const